File: benchmark/corpus/08_memory/f_two_layer_log.js
// The growth sits two helpers down: main -> remember -> append.
function append(log, entry) {
  log.push(entry); // DANGER
  return log;
}

function remember(log, i) {
  return append(log, "event " + i + " happened at tick " + Array.from({ length: 20 }, (_, k) => String(i * k % 97)).join(" "));
}

let log = [];
let i = 0;
while (true) {
  log = remember(log, i);
  i++;
}
console.log(log.length);

File: benchmark/corpus/08_memory/f_two_layer_log.py
# The growth sits two helpers down: main -> remember -> append.


def append(log, entry):
    log.append(entry)  # DANGER
    return log


def remember(log, i):
    return append(log, "event %d happened at tick %s" % (i, " ".join(str(i * k % 97) for k in range(20))))


log = []
i = 0
while True:
    log = remember(log, i)
    i += 1
print(len(log))
